packages/sources/ncfx/src/endpoint/crypto.ts

```diff
@@ -44,10 +73,13 @@ export const cryptoTransport = new WebSocketTransport<EndpointTypes>({
             logger.debug('Got logged in response, connection is ready')
             resolve()
           } else {
-            reject(new Error('Unexpected message after WS connection open'))
+            reject(
+              new Error(`Unexpected message after WS connection open: ${event.data.toString()}`),
```

Comment:
I recommend `JSON.stringify(event.data)` here instead, unless you know `data` is a primitive variable.

Reply:
So if I'm reading the typescript types for the websocket lib right, event.data has type `type Data = string | Buffer | ArrayBuffer | Buffer[];`. So I don't think it's safe to pass that directly to `JSON.stringify`. However, the `toString` method is guaranteed to exist on all 4 of those union types so we can use it. I'm also not 100% sure what you mean by primitive variable, but the data field should be exist (i.e. not null or undefined) according to the TS types. Also, we are already calling `event.data.toString()` just a few lines above so if there is any issue with it then it would blow up before it even gets to this line.